src/main/java/org/testng/reporters/FailedReporter.java

```diff
@@ -189,10 +162,8 @@ private void createXmlTest(ITestContext context, List<ITestNGMethod> methods, Xm
     Map<Class, Set<ITestNGMethod>> methodsMap= Maps.newHashMap();
 
     for (ITestNGMethod m : methods) {
-      Object[] instances= m.getInstances();
-      Class clazz= instances == null || instances.length == 0 || instances[0] == null
-          ? m.getRealClass()
-          : instances[0].getClass();
+      Object instances= m.getInstance();
```

Comment:
Is it important in the current fix? As I remember, `getInstance` should not be used and should be replaced by `getInstances`.